central/role/resources/list.go

```diff
@@ -89,10 +148,23 @@ func newResourceMetadata(name permissions.Resource, scope permissions.ResourceSc
 	return md
 }
 
-func newResourceMetadataWithFeatureFlag(name permissions.Resource, scope permissions.ResourceScope, flag features.FeatureFlag) permissions.ResourceMetadata {
+func newResourceMetadataWithReplacingResourceMetadata(name permissions.Resource, scope permissions.ResourceScope,
```

Comment:
would another name better explain the purpose of this function ?
```
func deprecatedResourceMetadata(name permissions.Resource, scope permissions.ResourceScope, replacement permissions.ResourceMetadata) permissions.ResourceMetadata
```
The remark applies to the variant with feature flag as well.

Reply:
Named the function `newDeprecatedResourceMetadata` to be in line with the other `new...` functions.